src/libcore/slice.rs

```diff
@@ -666,6 +666,60 @@ macro_rules! iterator {
                 let exact = diff / (if size == 0 {1} else {size});
                 (exact, Some(exact))
             }
+
+            #[inline]
+            fn count(self) -> usize {
```

Comment:
Yes but then this count won't be called (it can't because it needs to consume the iterator by value). Instead the default (slow) Iterator count will be called.